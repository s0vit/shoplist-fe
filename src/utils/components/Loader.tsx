import { alpha, useTheme } from 'src/shared/ui-kit';
import { Box, Paper } from 'src/shared/ui-kit';

import coinFlip from 'src/assets/giphy.webp';

const Loader = () => {
  const theme = useTheme();

  return (
    <Paper>
      <Box
        sx={{
          width: '100vw',
          height: '100vh',
          display: 'flex',
          flexDirection: 'column',
          justifyContent: 'center',
          alignItems: 'center',
          position: 'fixed',
          top: 0,
          left: 0,
          backgroundColor: alpha(theme.colors.cardBg, 0.5),
          backdropFilter: 'blur(5px)',
          zIndex: 9999,
        }}
      >
        <img src={coinFlip} alt="loading" />
      </Box>
    </Paper>
  );
};

export default Loader;
