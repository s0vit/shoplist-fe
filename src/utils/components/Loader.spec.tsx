import { render, screen, cleanup } from '@testing-library/react';
import Loader from 'src/utils/components/Loader.tsx';
import { describe, expect, it, afterEach, beforeEach } from 'vitest';

let renderResult: ReturnType<typeof render>;

const renderWithTheme = () => {
  return render(<Loader />);
};

describe('Loader', () => {
  beforeEach(() => {
    renderResult = renderWithTheme();
  });

  afterEach(() => {
    cleanup();
  });

  it('render loading img correctly', () => {
    const img = screen.getByAltText(/loading/i);
    expect(img).not.toBeNull();
    expect(img.tagName).toBe('IMG');
  });

  it('container with all styles exists', () => {
    const img = screen.getByAltText(/loading/i);
    const box = img.closest('div');
    expect(box).not.toBeNull();

    const styles = window.getComputedStyle(box!);
    expect(styles.width).toMatch(/\d+px/);
    expect(styles.height).toMatch(/\d+px/);
    expect(styles.position).toBe('fixed');
    expect(styles.zIndex).toBe('9999');
    expect(styles.backdropFilter).toBe('blur(5px)');
  });

  it('matches snapshot', () => {
    expect(renderResult.asFragment()).toMatchSnapshot();
  });
});
