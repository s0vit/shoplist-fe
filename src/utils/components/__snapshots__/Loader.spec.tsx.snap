// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Loader > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1ps6pg7-MuiPaper-root"
  >
    <div
      class="MuiBox-root css-2v5ii8"
    >
      <img
        alt="loading"
        src="/src/assets/giphy.webp"
      />
    </div>
  </div>
</DocumentFragment>
`;
